### ISSUE TITLE
What do we need for a 1.0 release

### ISSUE THREAD

**Issue by brooksph** [dahak-metagenomics/dahak#64]

**Comment by charlesreid1** [comment]
* command line interface - a friendly and more structured way to run Snakemake rules (c.f. [2018-snakemake-cli](https://github.com/ctb/2018-snakemake-cli))
* tests (covered on page 5 of the report) - regression tests, fixing any critical software versions, additional test metrics
* documentation - can take the form of markdown documents with examples, combined with interactive help provided by command line utility (sphinx might be overkill; if we did use it, it would just be as a style wrapper around a couple of markdown pages).

**Comment by charlesreid1** [comment]
A git clone of this repo currently takes upwards of 5 minutes on my machine. I think this is something we should solve _before_ a 1.0 release. Would you be open to that? If so, I'll start a thread where we can discuss options.

**Comment by brooksph** [comment]
Yes, we should address that. Please start an issue for that and feel free to add to the list.

**Comment by charlesreid1** [comment]
- [ ] create a `docs/` directory for the dahak sphinx documentation to live
- [ ] create a `cli/` or `taco/` directory for the dahak command line interface to live
- [ ] convert a few existing Snakefile workflows in the repository into the atomic rules-based format used in [dahak-taco](https://github.com/charlesreid1/dahak-taco/), so that they can be used from the command line

**Comment by charlesreid1** [comment]
Also @brooksph here is the taco "project" that I mentioned: https://github.com/charlesreid1/dahak-taco/projects

**Comment by charlesreid1** [comment]
Just to follow up here:

* We released version 1.0 beta of [dahak-taco](https://github.com/dahak-metagenomics/dahak-taco) on Friday 4/27
* We presented taco 1.0beta at the DIB Lab and got some good feedback on the design, so we're working on incorporating that. Also, it seemed like taco scratches an itch that other software doesn't, so there was solid interest in the tool from lab members. Good sign!
* We have solid documentation and have gotten good feedback on improvements we can make to the documentation
* We have structured repos to make testing easier (separating taco repo from workflow repos, enabling testing taco separately from testing individual workflows)
* We will be soliciting further rounds of feedback from other end users throughout the month of May in preparation for the 1.0 rollout.
